nuxeo-features/rest-api/nuxeo-rest-api-server/src/main/java/org/nuxeo/ecm/restapi/server/jaxrs/OAuth2Object.java

```diff
@@ -337,14 +509,15 @@ protected Response buildResponse(StatusType status, Object obj) throws IOExcepti
                        .build();
     }
 
-    protected void checkPermission() {
-        if (!hasPermission()) {
+    protected void checkPermission(String nxuser) {
+        if (!hasPermission(nxuser)) {
             throw new WebSecurityException("You do not have permissions to perform this operation.");
         }
     }
 
-    protected boolean hasPermission() {
-        return ((NuxeoPrincipal) getContext().getCoreSession().getPrincipal()).isAdministrator();
+    protected boolean hasPermission(String nxuser) {
+        NuxeoPrincipal principal = ((NuxeoPrincipal) getContext().getCoreSession().getPrincipal());
+        return principal.isAdministrator() || (nxuser == null ? false : principal.getName().equals(nxuser));
```

Comment:
Can you change by `nxuser.equals(principal.getName())`, as we know `nxuser` is not `null`.